chalk-solve/Cargo.toml

```diff
@@ -15,11 +15,16 @@ itertools = "0.9.0"
 petgraph = "0.5.0"
 rustc-hash = { version = "1.0.0" }
 
-chalk-macros = { version = "0.10.1-dev", path = "../chalk-macros" }
+chalk-engine-base = { version = "0.10.1-dev", path = "../chalk-engine-base" }
 chalk-derive = { version = "0.10.1-dev", path = "../chalk-derive" }
-chalk-engine = { version = "0.10.1-dev", path = "../chalk-engine" }
-chalk-ir = { version = "0.10.1-dev", path = "../chalk-ir" }
-chalk-rust-ir = { version = "0.10.1-dev", path = "../chalk-rust-ir" }
+chalk-engine = { version = "0.10.1-dev", path = "../chalk-engine", optional = true }
+chalk-ir = { version = "0.10.1-dev", path = "../chalk-ir", default-features = false }
 
 [dev-dependencies]
 chalk-integration = { version = "0.10.1-dev", path = "../chalk-integration" }
+
+[features]
+default = ["slg-solver", "recursive-solver"]
+
+slg-solver = ["chalk-engine", "chalk-ir/slg-solver"]
+recursive-solver = []
```

Comment:
I ran into https://github.com/rust-lang/cargo/issues/7916 while testing the feature flags, but I don't think it affects downstream users (like RA). The issue is that a `cargo check --no-default-features --features recursive-solver` in `chalk-solve` would still compile `chalk-engine` because the dev-dependency `chalk-integration` implicitly enables `chalk-ir`'s `slg-solver` feature. If the dev-dependency features are not unified, this issue goes away (tested on nightly with `-Z features=dev_dep`).

Reply:
Hmm. So what are the implications of this? We just always have to compile with both features?

Reply:
I think it works as expected for downstream crates (with `recursive-solver`, `chalk-engine` is not compiled and vice versa). However, when running commands in this repo like `cargo test`/`cargo check`, both solvers will get built.

Reply:
Yup, had some experience with it. This is okay.